src/org/loklak/harvester/strategy/BlockingKaizenHarvester.java

```diff
@@ -0,0 +1,61 @@
+package org.loklak.harvester.strategy;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import org.loklak.data.DAO;
+
+public class BlockingKaizenHarvester extends KaizenHarvester {
+
+    /**
+     * This class uses blocking queue to store queries which need to be processed.
+     */
+    public BlockingKaizenHarvester() {
+        super(new KaizenQueries() {
```

Comment:
Please just make this into an inner class or something. This looks ugly and unmaintainable.